### PR TITLE
#510 Remove plugin exports and update docs

### PR DESCRIPTION
**Pull request by ryanolee**
# What
Removing plugin exports as discussed in #510 

# Why 
So that plugin code is not too tightly coupled to custom plugin implementations

# How
Remove plugin exports and updated documentation 

# Notes
Note: This should not be merged until V6

### CODE REVIEW

**Comment by ryanolee** [comment]
Will fix the tests relating to this later today / tomorrow (time permitting 😆)

File: README.md [hunk 1]
```diff
@@ -204,20 +204,43 @@ Number, maximum amount of concurrent requests. Defaults to `Infinity`.
...
-    * [afterFinish](#afterfinish)
-    * [error](#error)
-    * [beforeRequest](#beforerequest)
-    * [afterResponse](#afterresponse)
-    * [onResourceSaved](#onresourcesaved)
-    * [onResourceError](#onresourceerror)
-    * [saveResource](#saveresource)
-    * [generateFilename](#generatefilename)
-    * [getReference](#getreference)
+- [website-scraper](#website-scraper)
```
Comment (s0ph1e) [inline README.md:207]: There are too many changes here, could you please leave only plugins-related items?

File: README.md [hunk 2]
```diff
@@ -204,20 +204,43 @@ Number, maximum amount of concurrent requests. Defaults to `Infinity`.
...
+        - [onResourceSaved](#onresourcesaved)
+        - [onResourceError](#onresourceerror)
+        - [generateFilename](#generatefilename)
+        - [getReference](#getreference)
+        - [saveResource](#saveresource)
+  - [result](#result)
+  - [Log and debug](#log-and-debug)
 
 ##### Built-in plugins
 Scraper has built-in plugins which are used by default if not overwritten with custom plugins. You can find them in [lib/plugins](https://github.com/website-scraper/node-website-scraper/tree/master/lib/plugins) directory or get them using
```
Comment (s0ph1e) [inline README.md:246]: I think the documentation also needs an update here

**Review by s0ph1e — CHANGES_REQUESTED** [review]
Hi @ryanolee

Thank you for creating the PR. Please fix the failing tests and update the readme according to the comments

**Comment by ryanolee** [comment]
Hi @s0ph1e ,
Thanks for the reply! I think the extra changes the readme files came from one of my IDE extensions that annoyingly seems to run on pre commit. Removed the change for now and fixed up the tests / docs. Not sure what we want to do with this PR as it probably does not need to go out until v6 of the scraper. 👀

**Comment by s0ph1e** [comment]
Hey @ryanolee,

Thank you for the quick changes. I'm going to merge the PR to master (master stands for current WIP for the next version, v5 code will be in branch 5.x)

**Pull request merged**